### PR TITLE
feat: Fetch `team_settings` data for Editor forms and adding Update fn

### PR DESCRIPTION
**Pull request by RODO94**
## What does this PR do?

_This is a cleaner branch from the work undertaken on this previous branch: #3319_ 

Continuing work on the ``team_settings`` implementation, this PR pulls in data from the database with a graphQL query and populates the corresponding form inputs.

This PR also uses the update mutation, similar to `team_themes` to update the form fields and database. 

This PR aligns with a planx-core change where the requests and types for a Team were altered to incorporate the new ``team_settings`` table.

[Reference to the planx-core changes](https://github.com/theopensystemslab/planx-core/pull/424)

> Important to note that this PR has not deleted any functions, methods, or types related to previous versions of `notifyPersonalisation` or the original ``settings`` column of _"teams"_ table. Thinking is to do this clear out after changes have been made and tested using the new ``team_settings`` table

### CODE REVIEW

File: editor.planx.uk/src/routes/teamSettings.tsx [hunk 1]
```diff
@@ -47,11 +47,11 @@ const teamSettingsRoutes = compose(
...
-              // {
-              //   name: "General",
-              //   route: "general",
-              //   Component: GeneralSettings,
-              // },
+              {
+                name: "General",
+                route: "general",
+                Component: GeneralSettings,
+              },
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/routes/teamSettings.tsx:54]: nit: I think I'd reorder this array to be General → Design → Team

File: editor.planx.uk/src/pages/FlowEditor/components/Settings/GeneralSettings/ContactForm.tsx [hunk 2]
```diff
@@ -20,9 +21,16 @@ export default function ContactForm({ formikConfig, onSuccess }: FormProps) {
...
     ...formikConfig,
     validationSchema: formSchema,
-    onSubmit(values, { resetForm }) {
-      onSuccess();
-      resetForm({ values });
+    onSubmit: async (values, { resetForm }) => {
+      const isSuccess = await useStore.getState().updateTeamSettings({
+        helpEmail: values.helpEmail,
+        helpOpeningHours: values.helpOpeningHours,
+        helpPhone: values.helpPhone,
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Settings/GeneralSettings/ContactForm.tsx:28]: `homepage` is missing here which means that changes to the URL are not persisted.

File: editor.planx.uk/src/pages/FlowEditor/components/Settings/GeneralSettings/BoundaryForm.tsx [hunk 3]
```diff
@@ -9,9 +10,14 @@ import { FormProps } from ".";
 export default function BoundaryForm({ formikConfig, onSuccess }: FormProps) {
   const formik = useFormik({
     ...formikConfig,
-    onSubmit(values, { resetForm }) {
-      onSuccess();
-      resetForm({ values });
+    onSubmit: async (values, { resetForm }) => {
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Settings/GeneralSettings/BoundaryForm.tsx:13]: We're still missing some basic validation on this field. I appreciate there's still work to be done on the boundary field, but if we're going to unhide the menu and allow Editors to access it we should either put this in place first, or hide this field.

image

**Review by DafyddLlyr — CHANGES_REQUESTED** [review]
Looking good - a lot simpler and cleaner following those naming updates 👍 

There's a few outstanding issues to address, please see comments. 

Also, I'd take a look at co-locating errors alongside their inputs. This will lead to a better user experience, and is required for accessibility reasons. This can be addressed here or in a follow up PR - whatever works best for you.

image

File: editor.planx.uk/src/pages/FlowEditor/components/Settings/GeneralSettings/BoundaryForm.tsx (hunk 3 above)
Comment (RODO94) [reply editor.planx.uk/src/pages/FlowEditor/components/Settings/GeneralSettings/BoundaryForm.tsx:13]: I'll do the error handling in another PR, hopefully make it a lot cleaner. Current ones were just to test my understanding of formik and yup

File: editor.planx.uk/src/routes/teamSettings.tsx (hunk 1 above)
Comment (RODO94) [reply editor.planx.uk/src/routes/teamSettings.tsx:54]: @DafyddLlyr do you think it is worth doing these types of changes as a final PR when error handling is all sorted?

Comment (RODO94) [reply editor.planx.uk/src/routes/teamSettings.tsx]: Thought about it again, and tit doesn't really make it difference, so I've added it to this PR

File: editor.planx.uk/src/pages/FlowEditor/components/Settings/GeneralSettings/BoundaryForm.tsx (hunk 3 above)
Comment (RODO94) [reply editor.planx.uk/src/pages/FlowEditor/components/Settings/GeneralSettings/BoundaryForm.tsx:13]: I have added basic validation now though

**Review by DafyddLlyr — APPROVED** [review]
Perfect! Looking great 😄 ✅ 

Agree that validation in the next PR will be the best way to go here 👍

**Comment by RODO94** [comment]
> Perfect! Looking great 😄 ✅
> 
> Agree that validation in the next PR will be the best way to go here 👍

Maybe I should move that last commit to a new PR then? @DafyddLlyr

**Comment by DafyddLlyr** [comment]
Sorry for not being clear, I think this should merge as-is, and we can tackle per-field validation next in a follow up PR.

**Comment by RODO94** [comment]
[Create `team_settings` table](https://trello.com/c/ttAMDUjk/2511-create-teamsettings-table)

**Comment by RODO94** [comment]
[Create `team_settings` table](https://trello.com/c/ttAMDUjk/2511-create-teamsettings-table)

**Pull request merged**